### DISCUSSION TITLE
Asynchronous TPDO

### DISCUSSION THREAD

**Discussion by SvanLeeuwen5** [canopen-python/canopen#337] [Q&A] [answered]
Hi all,

I was wondering if it is possible to transmit asynchronous TPDO messages based on internal events of the objects in the PDO mapping. 

I used transmission types 245 and 255. 
I set the inhibit time low and the event timer high, however the TPDO will only transmit on the event timer.
The internal data changes often because I use the Position Actual Value of the CiA 402 protocol. 

Hope someone can give me an answer
Thanks

**Comment by acolomb** [comment, marked as answer]
That's really a question for your device manufacturer.  TPDOs are sent from the device, the python-canopen script on your computer is not involved in the transmission, but only receives the PDO.  You can set the PDO transmission type through the API, something like `node.tpdo[1].trans_type = 254` followed by `node.tpdo[1].save()`.

**Reply by SvanLeeuwen5** [reply]
I was asking this question based on this comment:
https://github.com/christiansandberg/canopen/issues/166#issuecomment-551949792
He says that the PDO do not react to NMT messages and parameter changes, which is exactly my problem.
I also asked the manufacturer like you suggested so I'm curious what they will say.
